packages/open-next/src/core/routingHandler.ts

```diff
@@ -22,12 +22,17 @@ export interface MiddlewareOutputEvent {
   origin: Origin | false;
 }
 
+// Add the locale prefix to the regex so we correctly match the rawPath
+const optionalLocalePrefixRegex = !!RoutesManifest.locales.length
+  ? `^/(?:${RoutesManifest.locales.map((locale) => `${locale}/?`).join("|")})?`
```

Comment:
This cause an issue with 3 e2e test because of this, it should also include the default case (i.e. with no locales prefix)
![image](https://github.com/sst/open-next/assets/2964339/c66aceb7-28de-452d-b1db-c34ff209f264)
For the next data test it is because of this https://github.com/sst/open-next/blob/9d632d1bd5c3a8d7e7bbd13ae9daaca68da9284e/packages/tests-e2e/tests/pagesRouter/data.test.ts#L13 It ends now with `en.json`

Reply:
What would be the best approach here? Create a new pages app with i18n + additional e2e tests? 🤔 . It's expected that in this case it's `en.json` instead of `index.json` (but only if you have i18n settings).

It's a bit of a mess (on Next.js side)

Reply:
In a perfect world, yeah we should do something like that, but honestly just testing with `en.json` is probably enough